astropy/cosmology/flrw.py

```diff
@@ -104,17 +105,29 @@ class FLRW(Cosmology):
     documentation on :ref:`astropy-cosmology-fast-integrals`.
     """
 
+    H0 = Parameter(doc="Hubble constant as an `~astropy.units.Quantity` at z=0",
+                   unit="km/(s Mpc)")
+    Om0 = Parameter(doc="Omega matter; matter density/critical density at z=0")
+    Ode0 = Parameter(doc="Omega dark energy; dark energy density/critical density at z=0")
+    Tcmb0 = Parameter(doc="Temperature of the CMB as `~astropy.units.Quantity` at z=0",
+                      unit="Kelvin", fmt="0.4g")
+    Neff = Parameter(doc="Number of effective neutrino species")
+    m_nu = Parameter(doc="Mass of neutrino species",
+                     unit="eV", equivalencies=u.mass_energy(), fmt="")
+    Ob0 = Parameter(doc="Omega baryon; baryonic matter density/critical density at z=0")
+
     def __init__(self, H0, Om0, Ode0, Tcmb0=0.0*u.K, Neff=3.04, m_nu=0.0*u.eV,
                  Ob0=None, *, name=None, meta=None):
         super().__init__(name=name, meta=meta)
+        cls = self.__class__
 
         # all densities are in units of the critical density
-        self._Om0 = float(Om0)
+        self._Om0 = Parameter.validate(cls.Om0, float(Om0))
```

Comment:
See above comment: I think this would be clearer as `self.Om0 = float(Om0)` (and possibly the `float` check can be done in `validate`??). At least, I think parameters are always guaranteed to be set in `__init__`, correct, so a set-once scheme will always work?

(Possibly, with `parameters`, this setting could now be automated too, as is the case for `Representation`, but perhaps that is too much magic here...)

Reply:
See https://github.com/astropy/astropy/pull/12136#issuecomment-915618900. I think this problem is tricky and can be tackled at a later date.